File: java_junit5_idea/src/ExampleTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.fail;

public class ExampleTest {
    @Test
    public void 失敗するテスト() {
        assertEquals("expected", "actual", "失敗する");
    }

    enum Three {
        A, B, C
    }
    @Test
    public void inexhaustiveSwitch() {
        Three opt = Three.B;
        switch (opt) {
            case A:
                break;
            case B:
                fail();
                break;
        }
    }
}

File: kotlin_junit5_idea/src/example_test.kt
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.function.Executable
import org.junit.jupiter.api.Assertions.assertAll
import org.junit.jupiter.api.Assertions.assertEquals

internal class ExampleJUnit5KotlinTest {

    @Test
    fun test() {
        assertAll(
                Executable {
                    println("1st")
                    assertEquals(1, 0 + 1)
                },
                Executable {
                    println("2nd")
                    assertEquals(3, 0 + 1)
                },
                Executable {
                    println("3rd")
                    assertEquals(2, 1 + 1)
                }
        )
    }
}